### PR TITLE
Migrate to go modules

### PR DESCRIPTION
**Pull request by coolaj86**
Re: https://github.com/go-gitea/gitea/issues/5044

* [x] remove a few unused modules (bolt, gorilla)
* [x] `go mod init` and `go mod tidy` (adds `go.mod` and `go.sum`)
* [x] remove Gopkg.*
* [ ] update drone file? (doesn't look like there was much to do)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+vendor
```
Comment (jonasfranz) [inline .gitignore:1]: IMHO we shouldn't remove the vendor dir.
If a dependency or its host goes down, we can't build Gitea anymore.

**Review by jonasfranz — CHANGES_REQUESTED** [review]

**Comment by jonasfranz** [comment]
Please update the makefile in order to make sure that our CI tasks are going to work.

File: Makefile [hunk 2]
```diff
@@ -172,23 +172,6 @@ coverage:
 unit-test-coverage:
 	for PKG in $(PACKAGES); do $(GO) test -tags=sqlite -cover -coverprofile $$GOPATH/src/$$PKG/coverage.out $$PKG || exit 1; done;
 
-.PHONY: vendor
```
Comment (jonasfranz) [inline Makefile:175]: Don't remove the vendor and test-vendor tasks.

File: cmd/web.go [hunk 3]
```diff
@@ -21,7 +21,6 @@ import (
 	"code.gitea.io/gitea/routers/routes"
 
 	"github.com/Unknwon/com"
-	context2 "github.com/gorilla/context"
```
Comment (jonasfranz) [inline cmd/web.go:24]: Aren't these changes from https://github.com/go-gitea/gitea/pull/5068/files#diff-da541ef36e39344f2e12bddd91d363e2 ?

**Review by jonasfranz — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
Since Gitea depends on so many Go libraries. I would like to keep vendor to use `go mod vendor`.

**Comment by appleboy** [comment]
@coolaj86 Add build test in Drone

```yaml
  build-with-go-module:
    image: golang:1.11
    pull: true
    environment:
      - TAGS="bindata sqlite"
      - GO111MODULE=on
    commands:
      - make clean
      - make generate
      - make build
    when:
      event: [ push, tag, pull_request ]
```

**Comment by xor-gate** [comment]
As @lunny suggests indeed it is good to cache also in the `vendor` folder. Because when somebody decides to pull the plug out of a dependency you are screwed.

File: .drone.yml [hunk 4]
```diff
@@ -56,7 +56,7 @@ pipeline:
       event: [ push, tag, pull_request ]
 
   build-without-gcc:
-    image: golang:1.9
```
Comment (techknowlogick) [inline .drone.yml:59]: Can you keep this pegged at 1.9 as it is currently the oldest version of go that is supported.

Comment (techknowlogick) [reply .drone.yml:59]: (but the others can remain at 1.11 as we just need to ensure that 1.9 builds, but we will distribute the versions of gitea that are built with a more recent version)

**Comment by techknowlogick** [comment]
Closing due to #5743

**Pull request closed without merging**